src/Uno.UI.Runtime.Skia.Gtk/UI/Xaml/Controls/TextBoxViewExtension.cs

```diff
@@ -320,14 +327,34 @@ public void Select(int start, int length)
 			EnsureWidget(textBox);
 			if (_currentInputWidget is Entry entry)
 			{
-				textBox.UpdateFocusState(FocusState.Programmatic);
-				entry.SelectRegion(start_pos: start, end_pos: start + length);
+				if (textBox.FocusState == FocusState.Unfocused)
+				{
+					// Native control can't handle selection until it is part of visual tree.
+					// Use managed selection until then.
+					_requestedSelectionStart = textBox.Text.Length >= start ? start : textBox.Text.Length;
+					_requestedSelectionLength = textBox.Text.Length >= start + length ? length : textBox.Text.Length - start;
+				}
+				else
+				{
+					entry.SelectRegion(start_pos: start, end_pos: start + length);
+				}
 			}
 			// TODO: Handle TextView..
 		}
 
 		public int GetSelectionStart()
 		{
+			var textBox = _owner.TextBox;
+			if (textBox == null)
+			{
+				return 0;
+			}
+
+			if (textBox.FocusState == FocusState.Unfocused)
+			{
+				return _requestedSelectionStart ?? 0;
+			}
```

Comment:
What happens in the following scenario?

1. Text = "0123456"
2. Request selection start to be 3
3. Focus the control and move the cursor to the end.
4. Unfocus the control and call `GetSelectionStart`

I think it will return 3, but should return 7?

Reply:
I am clearing the selection when the native control shows up, so after that the selection info is coming from it directly. Not sure what UWP does in this scenario though - maybe we should cache the native controls' return values before the focus is lost

Reply:
> maybe we should cache the native controls' return values before the focus is lost

Yep that's what I meant. But should confirm what UWP does first certainly.